internal/store/build_result.go

```diff
@@ -270,9 +270,19 @@ type BuildState struct {
 	// This must be liberal: it's ok if this has too many files, but not ok if it has too few.
 	FilesChangedSet map[string]bool
 
-	// Whether there was a manual trigger
-	// TODO(nick): Replace this with a BuildReason
-	NeedsForceUpdate bool
+	// There are three kinds of triggers:
+	//
+	// 1) If a resource is in trigger_mode=TRIGGER_MODE_AUTO, then the resource auto-builds.
+	//    Pressing the trigger will always do a full image build.
+	//
+	// 2) If a resource is in trigger_mode=TRIGGER_MODE_MANUAL and there are no pending changes,
+	//    then pressing the trigger will do a full image build.
+	//
+	// 3) If a resource is in trigger_mode=TRIGGER_MODE_MANUAL, and there are pending changes,
+	//    then pressing the trigger will do a live_update
+	//
+	// We want to capture only case (1) and (2).
```

Comment:
```suggestion
	// This field indicates case 1 || case 2 -- i.e. that we should skip live_update, and force an image build (even if there are no changed files)
```
or something (w/o explanation of what this bool causes down the road, the explanation is less useful imo)

Reply:
done